tests/test_throttling.py

```diff
@@ -0,0 +1,43 @@
+import io
```

Comment:
would add a comment to this test, since it is based on debug logs (correct me if I am wrong, because of the -vvv) - it'll be flaky since if a line of **log** changes, this will break the test (although the logic might be working).
perhaps it is good to think about checking keepstate?